strawberry/fastapi/router.py

```diff
@@ -24,8 +24,15 @@
 
 class BaseContext:
     def __init__(self):
-        self.request: Union[Request, WebSocket] = None
-        self.background_tasks: BackgroundTasks = None
+        self.request: Optional[Union[Request, WebSocket]] = None
+        self.background_tasks: Optional[BackgroundTasks] = None
+        self.response: Optional[Response] = None
+
+
+CustomContext = Union[BaseContext, Dict[str, Any]]
+MergedContext = Union[
+    BaseContext, Dict[str, Union[Any, BackgroundTasks, Request, Response, WebSocket]]
+]
```

Comment:
1. @patrick91, do you know the best practices for type aliases? Top of file, or next to usage (between classes in this case)?
2. 
```suggestion
CustomContext: TypeAlias = Union[BaseContext, Dict[str, Any]]
MergedContext: TypeAlias = Union[
    BaseContext, 
    Dict[str, Union[Any, BackgroundTasks, Request, Response, WebSocket]],
]
```
3. What are the implications of `Union[Any, BackgroundTasks, Request, Response, WebSocket]]`? Is the `Any` supposed to be there? I'm unclear how it will affect the hints.

Reply:
If they're at the top then `BaseContext` isn't within their scope. Would doing:
```python
CustomContext = Union['BaseContext', Dict[str, Any]]
MergedContext = Union[
    'BaseContext', Dict[str, Union[Any, BackgroundTasks, Request, Response, WebSocket]]
]
```
be viable in this case? Or shall we move BaseContext to its own file, say `context.py` and import it?

Reply:
@BryceBeagle, the `Any` in that `Union` signifies the completely general type that the user can supply in their context dictionary as their custom value(s).

Reply:
@patrick91 should we be using (enforcing?) `from __future__ import annotation`-style forward references?

Reply:
uhm, let's not for now 😊 I think it is fine to use both styles, at least until we know what will happen to `from __future__ import annotation` :D